### ISSUE TITLE
Error make build - go-mod-core-contracts version

### ISSUE THREAD

**Issue by Salvocll** [edgexfoundry/device-sdk-go#385]
Hi,

I noticed that following the official guide to create the device-simple (official guide), once I run the make build, it goes to write to the file go.mod:

```
require (
github.com/edgexfoundry/device-sdk-go v1.0.0
github.com/edgexfoundry/go-mod-core-contracts v0.1.32
)
```

and for the versions ranging from v0.1.25 to v0.1.32 (go-mod-core-contracts) give the same compilation problems, while with previous versions not.

The compilation errors that are:

```
CGO_ENABLED=0 GO111MODULE=on go build -ldflags "-X github.com/edgexfoundry/device-simple.Version=1.1.0" -o cmd/device-simple/device-simple ./cmd/device-simple
# github.com/edgexfoundry/device-sdk-go/internal/clients
../../../../pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.0.0/internal/clients/init.go:202:58: cannot use endpoint (type *endpoint.Endpoint) as type clients.Endpointer in argument to metadata.NewAddressableClient:
	*endpoint.Endpoint does not implement clients.Endpointer (missing Fetch method)
../../../../pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.0.0/internal/clients/init.go:206:48: cannot use endpoint (type *endpoint.Endpoint) as type clients.Endpointer in argument to metadata.NewDeviceClient:
	*endpoint.Endpoint does not implement clients.Endpointer (missing Fetch method)
../../../../pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.0.0/internal/clients/init.go:210:62: cannot use endpoint (type *endpoint.Endpoint) as type clients.Endpointer in argument to metadata.NewDeviceServiceClient:
	*endpoint.Endpoint does not implement clients.Endpointer (missing Fetch method)
../../../../pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.0.0/internal/clients/init.go:214:62: cannot use endpoint (type *endpoint.Endpoint) as type clients.Endpointer in argument to metadata.NewDeviceProfileClient:
	*endpoint.Endpoint does not implement clients.Endpointer (missing Fetch method)
../../../../pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.0.0/internal/clients/init.go:221:46: cannot use endpoint (type *endpoint.Endpoint) as type clients.Endpointer in argument to coredata.NewEventClient:
	*endpoint.Endpoint does not implement clients.Endpointer (missing Fetch method)
../../../../pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.0.0/internal/clients/init.go:225:66: cannot use endpoint (type *endpoint.Endpoint) as type clients.Endpointer in argument to coredata.NewValueDescriptorClient:
	*endpoint.Endpoint does not implement clients.Endpointer (missing Fetch method)
Makefile:18: recipe for target 'cmd/device-simple/device-simple' failed
make: *** [cmd/device-simple/device-simple] Error 2
```

Instead, from version 0.1.0 to version 0.1.24, compile and generate the executable.

** Note: ** this is a past problem: https://github.com/edgexfoundry/device-sdk-go/issues/372

**Comment by cloudxxx8** [comment]
You need to use the latest version of device-service-sdk to meet the new change of go-core-contract
github.com/edgexfoundry/device-sdk-go v0.0.0-20191023092354-4a3b48e27207

**Comment by lyh0302** [comment]
你需要根据你下载的版本的go.mod文件去看下go-core-contract版本是多少，然后在go.mod里更改一下你的go-core-contract的版本号 就可以make build了

**Comment by mpalmier74** [comment]
Same problem here, using device-sdk-go branch origin/edinburgh or master

```
module github.com/edgexfoundry/device-simple

go 1.13

require (
        github.com/edgexfoundry/device-sdk-go v1.1.1
        github.com/edgexfoundry/go-mod-core-contracts v0.1.42
)
```

```
CGO_ENABLED=0 GO111MODULE=on go build -ldflags "-X github.com/edgexfoundry/device-simple.Version=0.0.1" -o cmd/device-simple/device-simple ./cmd/device-simple
# github.com/edgexfoundry/device-sdk-go/internal/clients
../../../../pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.1.1/internal/clients/init.go:203:58: cannot use endpoint (type *endpoint.Endpoint) as type interfaces.Endpointer in argument to metadata.NewAddressableClient:
        *endpoint.Endpoint does not implement interfaces.Endpointer (wrong type for Monitor method)
                have Monitor("github.com/edgexfoundry/go-mod-core-contracts/clients/types".EndpointParams, chan string)
                want Monitor("github.com/edgexfoundry/go-mod-core-contracts/clients/types".EndpointParams) chan string
../../../../pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.1.1/internal/clients/init.go:207:48: cannot use endpoint (type *endpoint.Endpoint) as type interfaces.Endpointer in argument to metadata.NewDeviceClient:
        *endpoint.Endpoint does not implement interfaces.Endpointer (wrong type for Monitor method)
                have Monitor("github.com/edgexfoundry/go-mod-core-contracts/clients/types".EndpointParams, chan string)
                want Monitor("github.com/edgexfoundry/go-mod-core-contracts/clients/types".EndpointParams) chan string
../../../../pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.1.1/internal/clients/init.go:211:62: cannot use endpoint (type *endpoint.Endpoint) as type interfaces.Endpointer in argument to metadata.NewDeviceServiceClient:
        *endpoint.Endpoint does not implement interfaces.Endpointer (wrong type for Monitor method)
                have Monitor("github.com/edgexfoundry/go-mod-core-contracts/clients/types".EndpointParams, chan string)
                want Monitor("github.com/edgexfoundry/go-mod-core-contracts/clients/types".EndpointParams) chan string
../../../../pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.1.1/internal/clients/init.go:215:62: cannot use endpoint (type *endpoint.Endpoint) as type interfaces.Endpointer in argument to metadata.NewDeviceProfileClient:
        *endpoint.Endpoint does not implement interfaces.Endpointer (wrong type for Monitor method)
                have Monitor("github.com/edgexfoundry/go-mod-core-contracts/clients/types".EndpointParams, chan string)
                want Monitor("github.com/edgexfoundry/go-mod-core-contracts/clients/types".EndpointParams) chan string
../../../../pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.1.1/internal/clients/init.go:219:57: cannot use endpoint (type *endpoint.Endpoint) as type interfaces.Endpointer in argument to general.NewGeneralClient:
        *endpoint.Endpoint does not implement interfaces.Endpointer (wrong type for Monitor method)
                have Monitor("github.com/edgexfoundry/go-mod-core-contracts/clients/types".EndpointParams, chan string)
                want Monitor("github.com/edgexfoundry/go-mod-core-contracts/clients/types".EndpointParams) chan string
../../../../pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.1.1/internal/clients/init.go:226:46: cannot use endpoint (type *endpoint.Endpoint) as type interfaces.Endpointer in argument to coredata.NewEventClient:
        *endpoint.Endpoint does not implement interfaces.Endpointer (wrong type for Monitor method)
                have Monitor("github.com/edgexfoundry/go-mod-core-contracts/clients/types".EndpointParams, chan string)
                want Monitor("github.com/edgexfoundry/go-mod-core-contracts/clients/types".EndpointParams) chan string
../../../../pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.1.1/internal/clients/init.go:230:66: cannot use endpoint (type *endpoint.Endpoint) as type interfaces.Endpointer in argument to coredata.NewValueDescriptorClient:
        *endpoint.Endpoint does not implement interfaces.Endpointer (wrong type for Monitor method)
                have Monitor("github.com/edgexfoundry/go-mod-core-contracts/clients/types".EndpointParams, chan string)
                want Monitor("github.com/edgexfoundry/go-mod-core-contracts/clients/types".EndpointParams) chan string
Makefile:18: recipe for target 'cmd/device-simple/device-simple' failed
make: *** [cmd/device-simple/device-simple] Error 2

```

**Comment by cloudxxx8** [comment]
we don't supported go-mod-core-contracts v0.1.42 in device-sdk-go v1.1.1
Please downgrade to
github.com/edgexfoundry/go-mod-core-contracts v0.1.31

**Comment by cloudxxx8** [comment]
To use master (the latest change) of device-sdk-go, it also only supports go-mod-core-contracts v0.1.36
Please use the timestamp instead of version number for master (not v1.1.1, because it is fuji)

**Comment by cloudxxx8** [comment]
fixed by #447
